MinecraftClient/Inventory/Container.cs

```diff
@@ -153,5 +153,38 @@ public static ContainerType GetContainerType(int typeID)
                 default: return ContainerType.Unknown;
             }
         }
+
+        /// <summary>
+        /// Search an item in the container
+        /// </summary>
+        /// <param name="itemType">The item to search</param>
+        /// <returns>An array of slot ID</returns>
+        public int[] SearchItem(ItemType itemType)
+        {
+            List<int> result = new List<int>();
+            if (Items != null)
+            {
+                foreach (var item in Items)
+                {
+                    if (item.Value.Type == itemType)
+                        result.Add(item.Key);
+                }
+            }
+            return result.ToArray();
+        }
+
+        public int[] GetEmpytSlot()
```

Comment:
`GetEmpytSlot` ▶️ `GetEmpytSlots`
```
/// <summary>
/// List empty slots in the container
/// </summary>
/// <returns>An array of slot ID</returns>
```
Also depending on the container type, some empty slots cannot be used e.g. armor slots. This might cause issues.